src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,84 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { createComment } from '../api/comments';
+import { Post } from '../types/Post';
+import { useComments } from './Contexts/CommentsContext';
+
+type Props = {
+  selectedPost: Post,
+};
+
+const initialCommentForm = {
+  name: '',
+  email: '',
+  commentText: '',
+};
+
+export const NewCommentForm: React.FC<Props> = ({ selectedPost }) => {
+  const { setComments } = useComments();
+
+  const [commentForm, setCommentForm] = useState(initialCommentForm);
+  const { name, email, commentText } = commentForm;
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsSubmitted(true);
+    if (name && email && commentText) {
+      setIsLoading(true);
+      const newComment = await createComment({
+        name,
+        email,
+        body: commentText,
+        postId: selectedPost.id,
+      });
+
+      try {
+        setComments(prev => [...prev, newComment]);
+      } catch {
+        // eslint-disable-next-line no-console
+        console.info('Something went wrong.');
+      }
+
+      setCommentForm(prev => ({ ...prev, commentText: '' }));
+      setIsSubmitted(false);
+      setIsLoading(false);
+    }
+  };
+
+  const onClear = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setCommentForm(initialCommentForm);
+    setIsSubmitted(false);
+  };
+
+  const commentHasError = !commentText && isSubmitted;
+  const emailHasError = !email && isSubmitted;
+  const nameHasError = !name && isSubmitted;
+
+  const handleChangeName = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCommentForm(prev => ({ ...prev, name: event.target.value }));
+  };
+
+  const handleChangeEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCommentForm(prev => ({ ...prev, email: event.target.value }));
+  };
+
+  const handleChangeComment = (
+    event: React.ChangeEvent<HTMLTextAreaElement>,
+  ) => {
+    setCommentForm(prev => ({ ...prev, commentText: event.target.value }));
+  };
```

Comment:
You can combine them into one handler. Input name you can get from `event.target`. Something like this

```javascript
const handleChange = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
  ) => {
    const { name, value } = event.target;
    setCommentForm(prev => ({ ...prev, [name]: value }));
  };
```